src/app/content/highlights/utils/paginationUtils.spec.ts

```diff
@@ -1,32 +1,53 @@
 import { HighlightColorEnum } from '@openstax/highlighter/dist/api';
 import { treeWithUnits } from '../../../../test/trees';
+import { resetModules } from '../../../../test/utils';
 import { findArchiveTreeNode } from '../../utils/archiveTreeUtils';
 import {
   filterCountsPerSourceByColorFilter,
   filterCountsPerSourceByLocationFilter,
-  getNextPageSources
 } from './paginationUtils';
 
 describe('getNextPageSources', () => {
-  it('gets one page id if it has enough records left to fill a response', () => {
-    expect(getNextPageSources({
-      page1: {[HighlightColorEnum.Green]: 10},
-      page2: {[HighlightColorEnum.Pink]: 10},
-    }, treeWithUnits, 10)).toEqual(['page1']);
+  beforeEach(() => {
+    resetModules();
   });
 
-  it('fails over to next page if more records are needed', () => {
-    expect(getNextPageSources({
-      page1: {[HighlightColorEnum.Green]: 5},
-      page2: {[HighlightColorEnum.Pink]: 10},
-    }, treeWithUnits, 10)).toEqual(['page1', 'page2']);
+  describe('under resource limit' , () => {
+    it('gets one page id if it has enough records left to fill a response', () => {
+      const {getNextPageSources} = require('./paginationUtils');
```

Comment:
I wasn't able to mock the `maxResourcesPerFetch` for one test only any other way. I'm not sure what am i doing wrong.

Reply:
you're doing it right. because regular imports are cached and you can't modify an exported constant by reference you have to reset and require it for each test you run. you can do it in a `beforeEach` to reduce duplication, but in general you're doing it right already.